### ISSUE TITLE
[Feature Request] Avoid needless messaging

### ISSUE THREAD

**Issue by bershanskiy** [darkreader/darkreader#9212]
Currently, the extension state is always relayed via background context instead of being retrieved from `chrome.storage` directly. The persistence of background context dictated this original design: all data was stored in globals or instances of a particular class. With non-persistent background, this design is less convenient because effectively this happens:
 - CS/UI context requests data via `sendMessage`
 - browser creates/activates background context
 - background context has to initialize itself and retrieves _a lot_ of data from storage via `chrome.storage`
 - background context gets to `onMessage` handler and picks out a tiny bit of info which was actually requested originally
 - background context calls `sendMessage` (or sometimes `sendResponse`) with the small amount of data
 - browser serializes message and relays it to CS/UI
 - CS/UI finally receives the required data, deserializes it, and uses it

Here is the diagram of the whole process:
```
+-------+                     +---------+                   +----+                        +---------+
| CS/UI | ---(sendMessage)--> | Browser | ---(onMessage)--> | BG | ---(chrome.storage)--> | Browser |
+-------+                     +---------+                   +----+                        +---------+
                                                                                               |
+-------+                   +---------+                     +----+                             |
| CS/UI | <--(onMessage)--- | Browser | <--(sendMessage)--- | BG | <---------------------------+
+-------+                   +---------+                     +----+
```

This is very convoluted and relatively inefficient since it spawns a separate process for the background context. Instead, CS/UI can call `chrome.storage` APIs directly like so:
```
+-------+                        +---------+
| CS/UI | ---(chrome.storage)--> | Browser |
+-------+                        +---------+
    ^                                  |
    +----------------------------------+
```

For reference, this is the recommended design of MV3/non-persistent extensions, which is officially recommended by Google. We avoided migrating to this design earlier because it was potentially disruptive. However, we can avoid the old design for all new keys like one in #9167.

**Comment by alexanderby** [comment]
Yes, this is what I actually did in the Safari version and other new extensions. Using the `chrome.storage.onChange` callback instead of messaging.

**Comment by bershanskiy** [comment]
I plan to fix this by making `StateManager` aware of `chrome.storage.onChanged` event and using it more consistently.

Part 1 of this refactor: #9271.
